### DISCUSSION TITLE
How to keep only text when clicking usernames (usercard text only)

### DISCUSSION THREAD

**Discussion by RazorMid** [Chatterino/chatterino2#5423] [Q&A] [answered]
I switched from Chatty to Chatterino for a while now and I like it a lot, however, for my needs, I would like to know if the following option is possible and how to achieve it. Basically, when I'm clicking some username in the chat and the usercard popup opens I would like to see only the text and nothing more as the window is too large. 

Here's an example of what I want to see from this 
![0a759feffb08f2c50bb9567f763f9caf](https://github.com/Chatterino/chatterino2/assets/39539652/2d476611-295c-4040-9262-a1e7cc72e546)
 to this only 
![cff150669b7f90eddf0377b4f443a11e](https://github.com/Chatterino/chatterino2/assets/39539652/90f4d75f-d1ff-4741-97b8-b47ae60f4e74) 
and eventually I would like to make the window even smaller in terms of height. 

Here is how Chatty's displaying messages from individuals 
![304157da28eac69ab0edb8083ef7a211](https://github.com/Chatterino/chatterino2/assets/39539652/b7092cf6-ade8-469f-84c6-36331bab5cd5)
 and as you can see the window is very small. Since I am tracking specific messages from multiple users I need to open multiple windows and if they're too large they're taking up all my screen so I am trying to have the text only being displayed. Thank you.

**Comment by pajlada** [comment]
Would being able to collapse a usercard into something like this be sufficient?
![usercard-collapsed-1](https://github.com/Chatterino/chatterino2/assets/962989/ab2b542b-a97a-4d0a-8bb6-89676f0cb8c6)

**Reply by RazorMid** [reply]
Thank you for answering, @pajlada. I would still like to get rid of the rest of the options if possible and only keep the window title, the minimize/maximize and close buttons and the text window basically. Something like this:
![aaaa](https://github.com/Chatterino/chatterino2/assets/39539652/22a525fd-20a7-4f11-9eab-fcf4df4b5841)

**Reply by pajlada** [reply]
What about only keeping the username & user id? this means it would still look good & be functional on systems without title bars

For reference, this is what the usercard looks like to me:
![image](https://github.com/Chatterino/chatterino2/assets/962989/a0091492-888e-4c68-8f26-2b5f49def6a3)

so being able to collapse it like this would leave the most relevant information still accessible
![image](https://github.com/Chatterino/chatterino2/assets/962989/c54741c1-51a4-48df-96b4-e7e70b896273)

**Reply by pajlada** [reply]
btw my counter-suggestions are just a way to figure out what would work & what wouldn't work. I'm assuming even after my last suggestion, your preference would still be:
1. chat
2. chat + username/id
3. chat + username/id + block/ignore actions
4. current look

correct me if I'm wrong

**Comment by RazorMid** [comment]
Thank you for the follow-up @pajlada, the second image you posted now is close to what I am trying to achieve, although no user avatar image should be displayed. As per your last reply, 2 is what I am trying to achieve so chat+username (ID is not needed really, I would rather have the close and minimize buttons for the popup window there instead as part of the title bar).

I don't need the block/ignore or moderator action buttons either, the image I previously attached is exactly the look I am trying to achieve.

**Comment by RazorMid** [comment]
How would I be able to collapse the usercard in this case?

**Reply by pajlada** [reply]
In this theoretical case, you'd have an arrow next to the avatar or username, allowing you to collapse things

**Reply by RazorMid** [reply]
> In this theoretical case, you'd have an arrow next to the avatar or username, allowing you to collapse things

Will removing that piece of information be easier than simply adding a collapse arrow? I wouldn't need that information for the future regardless. If yes, which part do I need to remove from the source code to achieve that and remove the avatar and the rest of the info?

**Reply by pajlada** [reply, marked as answer]
If you're interested in making a custom build that removes the information you're not interested in, it would be easier to just delete that information in the first place. You'd want to look in `src/widgets/dialogs/UserInfoPopup.cpp`, specifically towards the bottom of its constructor (starts around line 256) - it's what's responsible for initializing its UI. You should be able to just delete the code that adds things like follower info etc

**Reply by RazorMid** [reply]
> If you're interested in making a custom build that removes the information you're not interested in, it would be easier to just delete that information in the first place. You'd want to look in `src/widgets/dialogs/UserInfoPopup.cpp`, specifically towards the bottom of its constructor (starts around line 256) - it's what's responsible for initializing its UI. You should be able to just delete the code that adds things like follower info etc

Thank you very much for your help @pajlada, I will give this one a go next week as I need to install the required stuff to compile the software.

**Reply by pajlada** [reply]
Sounds good - if you run into any issues setting up the development environment or issues when you start running your modified build, feel free to poke us here!

**Reply by constellbell25** [reply]
thanks
